packages/ui-components/src/components/SearchInput/SearchInput.component.tsx

```diff
@@ -0,0 +1,226 @@
+    case "hero":
+      return `${wrapperBaseStyles} jn-w-full`
+    default:
+      return `${wrapperBaseStyles} jn-w-auto`
+  }
+}
+
+const searchStyles = (variant: "rounded" | "hero" | "default"): string => {
+  const searchBaseStyles = `
+    jn-bg-theme-textinput
+    jn-text-theme-high
+    jn-shadow
+    jn-w-full
+    focus:jn-outline-none
+    focus:jn-ring-2
+    focus:jn-ring-theme-focus
+    disabled:jn-cursor-not-allowed
+    disabled:jn-opacity-50
+  `
+
+  const roundedStyles = `
+    jn-rounded-full 
+    focus:jn-rounded-full
+  `
+  switch (variant) {
+    case "rounded":
+      return `${searchBaseStyles} ${roundedStyles} jn-text-base jn-w-auto jn-pl-3 jn-pr-16 jn-py-1`
+    case "hero":
+      return `${searchBaseStyles} ${roundedStyles} jn-text-lg jn-w-full jn-pl-6 jn-pr-20 jn-py-2.5`
+    default:
+      return `${searchBaseStyles} jn-rounded jn-text-base jn-leading-4 jn-pl-4 jn-pr-16 jn-py-2.5`
+  }
+}
+
+const iconWrapperStyles = (variant: "rounded" | "hero" | "default"): string => {
+  switch (variant) {
+    case "rounded":
+      return `jn-absolute jn-inline-flex jn-right-3 jn-top-1`
+    case "hero":
+      return `jn-absolute jn-inline-flex jn-right-5`
+    default:
+      return `jn-absolute jn-inline-flex jn-right-3 jn-top-2`
+  }
+}
+
+const clearIconStyles = (variant: "rounded" | "hero" | "default"): string => {
+  switch (variant) {
+    case "rounded":
+      return `jn-mr-2`
+    case "hero":
+      return `jn-mr-2.5`
+    default:
+      return `jn-mr-2`
+  }
+}
+
+const clearIconSize = (variant: "rounded" | "hero" | "default"): string => {
+  switch (variant) {
+    case "hero":
+      return "24"
+    default:
+      return "18"
+  }
+}
+
+/**
+ * A SearchInput is a controlled input component for searching.
+ * It provides a text field to enter a search query and optional clear and search icons.
+ * Three styling variants are supported: "rounded", "hero", and "default".
+ */
+
+export const SearchInput: React.FC<SearchInputProps> = ({
+  value = "",
+  name = "search",
+  variant = "default",
+  disabled = false,
+  clear = true,
+  onSearch,
+  onChange,
+  onClick,
+  onKeyPress,
+  onClear,
+  autoComplete = "off",
+  placeholder = "Search…",
+  className = "",
+  ...props
+}) => {
+  const [val, setValue] = useState(value)
+
+  useEffect(() => {
+    setValue(value)
+  }, [value])
+
+  const handleInputChange = (event: ChangeEvent<HTMLInputElement>): void => {
+    setValue(event.target.value)
+    onChange && onChange(event)
+  }
+
+  const handleKeyPress = (event: KeyboardEvent<HTMLInputElement>): void => {
+    if (event.key === "Enter" && onSearch) onSearch(val)
+    onKeyPress && onKeyPress(event)
+  }
+
+  const handleSearchClick: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement> = (event) => {
+    onSearch && onSearch(val)
+    onClick && onClick(event)
+  }
+
+  const handleClearClick: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement> = (event) => {
```

Comment:
I'd suggest to wrap all event handlers `handleInputChange`, ` handleKeyPress`, `handleSearchClick` and `handleClearClick` inside `useCallback` hook otherwise react will have to recreate those methods on each re-render.